### PR TITLE
python3Packages.python-telegram-bot: 13.5 -> 13.6

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 13.6

Change log: https://github.com/python-telegram-bot/python-telegram-bot/releases/tag/v13.6

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126258` at 36048062 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.ignite</li>
    <li>python38Packages.pytorchWithCuda</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.spacy-transformers</li>
    <li>python38Packages.tensorflowWithCuda</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.pytorchWithCuda</li>
    <li>python39Packages.tensorflowWithCuda</li>
  </ul>
</details>
<details>
  <summary>35 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/5be3e3a88e1125d13b00ee0b38b6dcd4">mirage-im</a></li>
    <li><a href="https://gist.github.com/533545564a1fe857956cec62285caffa">python38Packages.angr</a></li>
    <li><a href="https://gist.github.com/3ac465465a4b0a618ca19163df708c7a">python38Packages.angrop</a></li>
    <li><a href="https://gist.github.com/79581e37b8953c781d8a1574c03d65cf">python38Packages.cirq</a></li>
    <li><a href="https://gist.github.com/e20650f3e9dfdbbbf991572c3ed8e467">python38Packages.cirq-google</a></li>
    <li><a href="https://gist.github.com/3756e1a94c1dd686803201dfd90f66a4">python39Packages.angr</a></li>
    <li><a href="https://gist.github.com/a03ca7f3dd68171fa36f21f2429f8fb4">python39Packages.angrop</a></li>
    <li><a href="https://gist.github.com/59627028ae636eb720398907898b613f">python39Packages.cirq-google</a></li>
    <li><a href="https://gist.github.com/981bbaac2d57a00a30a6ced9e89c3170">python39Packages.google-cloud-asset</a></li>
    <li><a href="https://gist.github.com/b6ab3b524cbca91a7d89febc433c4eb3">python39Packages.google-cloud-automl</a></li>
    <li><a href="https://gist.github.com/c4865d67ba51212070c084b934d53e77">python39Packages.google-cloud-bigquery</a></li>
    <li><a href="https://gist.github.com/d762e155a9edb813240f9556b47e26f8">python39Packages.google-cloud-bigquery-datatransfer</a></li>
    <li><a href="https://gist.github.com/23467e93ff9116c9ca091c50180160a1">python39Packages.google-cloud-container</a></li>
    <li><a href="https://gist.github.com/8b0babc31f98ea12757dc47bd47434d9">python39Packages.google-cloud-dataproc</a></li>
    <li><a href="https://gist.github.com/4d1859b62f82107aa9fc6f37823fe237">python39Packages.google-cloud-datastore</a></li>
    <li><a href="https://gist.github.com/49167cd425ff1d3d782e9b44b05c4d43">python39Packages.google-cloud-dlp</a></li>
    <li><a href="https://gist.github.com/694388b792dad494bc4022fcd7ea73a0">python39Packages.google-cloud-error-reporting</a></li>
    <li><a href="https://gist.github.com/a79e838425795a4401bfa017b2e620a9">python39Packages.google-cloud-iam</a></li>
    <li><a href="https://gist.github.com/e967210eef55b057204f8650488cd962">python39Packages.google-cloud-iot</a></li>
    <li><a href="https://gist.github.com/fef2f6a10c5d5969b69ec03254e73322">python39Packages.google-cloud-kms</a></li>
    <li><a href="https://gist.github.com/17e9c562c91529348044e8e07a904672">python39Packages.google-cloud-language</a></li>
    <li><a href="https://gist.github.com/e1e88cd5503c013bac34642523e90877">python39Packages.google-cloud-monitoring</a></li>
    <li><a href="https://gist.github.com/e738738ab0ec4a7bf2d689a43d8757d2">python39Packages.google-cloud-os-config</a></li>
    <li><a href="https://gist.github.com/1e79db437358a693c9ef49a1c44fbec3">python39Packages.google-cloud-pubsub</a></li>
    <li><a href="https://gist.github.com/c89de36f6aaa0179a4e0f2419ba17b6e">python39Packages.google-cloud-redis</a></li>
    <li><a href="https://gist.github.com/f6e668b25638e3ad6ae62e38cf23de64">python39Packages.google-cloud-secret-manager</a></li>
    <li><a href="https://gist.github.com/bef63543212f70704f98fd3b16f2a474">python39Packages.google-cloud-securitycenter</a></li>
    <li><a href="https://gist.github.com/cbf527b12cfc90fd852e941e5c2bdf88">python39Packages.google-cloud-spanner</a></li>
    <li><a href="https://gist.github.com/7d042ebf30b40b22165088bf6f0a6322">python39Packages.google-cloud-speech</a></li>
    <li><a href="https://gist.github.com/6e66c2d9a673b64b9a9fee928acb9685">python39Packages.google-cloud-storage</a></li>
    <li><a href="https://gist.github.com/488a1cc93f452d3ed42035a5e70d958c">python39Packages.google-cloud-tasks</a></li>
    <li><a href="https://gist.github.com/efe8bbf5cb5d963964d4e4b2fa1464b3">python39Packages.google-cloud-texttospeech</a></li>
    <li><a href="https://gist.github.com/9f6d7484037cb62ba7d6af70d440a355">python39Packages.google-cloud-translate</a></li>
    <li><a href="https://gist.github.com/1521089b2434d5029c98d4def63e612b">python39Packages.google-cloud-vision</a></li>
    <li><a href="https://gist.github.com/cad05dfc205d85e8304dbcb38f024241">python39Packages.google-cloud-websecurityscanner</a></li>
  </ul>
</details>
<details>
  <summary>55 packages skipped due to time constraints:</summary>
  <ul>
    <li>home-assistant</li>
    <li>kmymoney</li>
    <li>python38Packages.arviz</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.dm-sonnet</li>
    <li>python38Packages.edward</li>
    <li>python38Packages.graph_nets</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>132 packages built successfully:</summary>
  <ul>
    <li>beancount (python38Packages.beancount)</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>dvc-with-remotes</li>
    <li>fava</li>
    <li>gcalcli</li>
    <li>git-annex-remote-googledrive</li>
    <li>ledger2beancount</li>
    <li>lieer</li>
    <li>matrix-commander</li>
    <li>mopidy-podcast</li>
    <li>mopidy-youtube</li>
    <li>opsdroid</li>
    <li>pantalaimon</li>
    <li>patroni</li>
    <li>python38Packages.beancount_docverif</li>
    <li>python38Packages.cachetools</li>
    <li>python38Packages.carbon</li>
    <li>python38Packages.chirpstack-api</li>
    <li>python38Packages.claripy</li>
    <li>python38Packages.coapthon3</li>
    <li>python38Packages.drivelib</li>
    <li>python38Packages.gcsfs</li>
    <li>python38Packages.goobook</li>
    <li>python38Packages.google-api-core</li>
    <li>python38Packages.google-api-python-client</li>
    <li>python38Packages.google-auth</li>
    <li>python38Packages.google-auth-httplib2</li>
    <li>python38Packages.google-auth-oauthlib</li>
    <li>python38Packages.google-cloud-access-context-manager</li>
    <li>python38Packages.google-cloud-asset</li>
    <li>python38Packages.google-cloud-automl</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.google-cloud-bigquery-datatransfer</li>
    <li>python38Packages.google-cloud-bigtable</li>
    <li>python38Packages.google-cloud-container</li>
    <li>python38Packages.google-cloud-core</li>
    <li>python38Packages.google-cloud-dataproc</li>
    <li>python38Packages.google-cloud-datastore</li>
    <li>python38Packages.google-cloud-dlp</li>
    <li>python38Packages.google-cloud-dns</li>
    <li>python38Packages.google-cloud-error-reporting</li>
    <li>python38Packages.google-cloud-firestore</li>
    <li>python38Packages.google-cloud-iam</li>
    <li>python38Packages.google-cloud-iot</li>
    <li>python38Packages.google-cloud-kms</li>
    <li>python38Packages.google-cloud-language</li>
    <li>python38Packages.google-cloud-logging</li>
    <li>python38Packages.google-cloud-monitoring</li>
    <li>python38Packages.google-cloud-org-policy</li>
    <li>python38Packages.google-cloud-os-config</li>
    <li>python38Packages.google-cloud-pubsub</li>
    <li>python38Packages.google-cloud-redis</li>
    <li>python38Packages.google-cloud-resource-manager</li>
    <li>python38Packages.google-cloud-runtimeconfig</li>
    <li>python38Packages.google-cloud-secret-manager</li>
    <li>python38Packages.google-cloud-securitycenter</li>
    <li>python38Packages.google-cloud-spanner</li>
    <li>python38Packages.google-cloud-speech</li>
    <li>python38Packages.google-cloud-storage</li>
    <li>python38Packages.google-cloud-tasks</li>
    <li>python38Packages.google-cloud-testutils</li>
    <li>python38Packages.google-cloud-texttospeech</li>
    <li>python38Packages.google-cloud-trace</li>
    <li>python38Packages.google-cloud-translate</li>
    <li>python38Packages.google-cloud-videointelligence</li>
    <li>python38Packages.google-cloud-vision</li>
    <li>python38Packages.google-cloud-websecurityscanner</li>
    <li>python38Packages.google-resumable-media</li>
    <li>python38Packages.gspread</li>
    <li>python38Packages.kubernetes</li>
    <li>python38Packages.locationsharinglib</li>
    <li>python38Packages.matrix-nio</li>
    <li>python38Packages.oauthenticator</li>
    <li>python38Packages.openshift</li>
    <li>python38Packages.pathy</li>
    <li>python38Packages.premailer</li>
    <li>python38Packages.py-air-control</li>
    <li>python38Packages.py-air-control-exporter</li>
    <li>python38Packages.pydrive</li>
    <li>python38Packages.pysmappee</li>
    <li>python38Packages.python-telegram-bot</li>
    <li>python38Packages.uproot3</li>
    <li>python38Packages.weboob</li>
    <li>python38Packages.woob</li>
    <li>python39Packages.beancount</li>
    <li>python39Packages.beancount_docverif</li>
    <li>python39Packages.cachetools</li>
    <li>python39Packages.carbon</li>
    <li>python39Packages.chirpstack-api</li>
    <li>python39Packages.claripy</li>
    <li>python39Packages.coapthon3</li>
    <li>python39Packages.drivelib</li>
    <li>python39Packages.gcsfs</li>
    <li>python39Packages.goobook</li>
    <li>python39Packages.google-api-core</li>
    <li>python39Packages.google-api-python-client</li>
    <li>python39Packages.google-auth</li>
    <li>python39Packages.google-auth-httplib2</li>
    <li>python39Packages.google-auth-oauthlib</li>
    <li>python39Packages.google-cloud-access-context-manager</li>
    <li>python39Packages.google-cloud-bigtable</li>
    <li>python39Packages.google-cloud-core</li>
    <li>python39Packages.google-cloud-dns</li>
    <li>python39Packages.google-cloud-firestore</li>
    <li>python39Packages.google-cloud-logging</li>
    <li>python39Packages.google-cloud-org-policy</li>
    <li>python39Packages.google-cloud-resource-manager</li>
    <li>python39Packages.google-cloud-runtimeconfig</li>
    <li>python39Packages.google-cloud-testutils</li>
    <li>python39Packages.google-cloud-trace</li>
    <li>python39Packages.google-cloud-videointelligence</li>
    <li>python39Packages.google-resumable-media</li>
    <li>python39Packages.gspread</li>
    <li>python39Packages.kubernetes</li>
    <li>python39Packages.locationsharinglib</li>
    <li>python39Packages.matrix-nio</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.openshift</li>
    <li>python39Packages.premailer</li>
    <li>python39Packages.py-air-control</li>
    <li>python39Packages.py-air-control-exporter</li>
    <li>python39Packages.pydrive</li>
    <li>python39Packages.pysmappee</li>
    <li>python39Packages.python-telegram-bot</li>
    <li>python39Packages.uproot3</li>
    <li>python39Packages.weboob</li>
    <li>python39Packages.woob</li>
    <li>s3ql</li>
    <li>wal_e</li>
    <li>weechatScripts.weechat-matrix</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `isPy3k`.
Near pkgs/development/python-modules/python-telegram-bot/default.nix:9:3:
```
  |
9 | , isPy3k
  |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Review by pingiun — CHANGES_REQUESTED** [review]
Could you move the update to cachetools to a different PR and add a line similar to `--replace 'APScheduler==3.6.3' 'APScheduler'` but for cachetools?

In my testing ptb works with this older version of cachetools. The upstream maintainers pin this dependency because they don't want to officially support any other versions.

**Comment by pingiun** [comment]
Also note that there is the unused argument of `isPy3k`, as @r-rmcgibbo said

File: pkgs/development/python-modules/python-telegram-bot/default.nix [hunk 1]
```diff
@@ -1,27 +1,30 @@
 { lib
 , APScheduler
 , buildPythonPackage
+, cachetools
 , certifi
 , decorator
 , fetchPypi
 , future
 , isPy3k
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-telegram-bot/default.nix:9]:
```suggestion
```

**Review by mweinelt — COMMENTED** [review]
Home Assistant tests are passing.

**Comment by mweinelt** [comment]
Merged into my local working tree for python-unstable. Thank you.

> Also note that there is the unused argument of isPy3k, as @r-rmcgibbo said

Dropped

> Could you move the update to cachetools to a different PR and add a line similar to --replace 'APScheduler==3.6.3' 'APScheduler' but for cachetools?

Please open a separate pull request if you think this should be done.

**Comment by pingiun** [comment]
Why was this closed? I would like to see this change in master

**Comment by mweinelt** [comment]
@pingiun I picked it into #127419.

**Comment by pingiun** [comment]
@mweinelt what's the reasoning in collecting these changes?

**Comment by mweinelt** [comment]
https://github.com/NixOS/nixpkgs/pull/126258#issuecomment-857354198 looked like lots of breakage.
